package/gluon-mesh-batman-adv/files/lib/netifd/proto/gluon_bat0.sh

```diff
@@ -29,10 +29,14 @@ proto_gluon_bat0_renew() {
 
 	lock /var/lock/gluon_bat0.lock
 
-	ubus call network.interface dump | jsonfilter \
-		-e "@.interface[@.proto='gluon_mesh' && @.up=true].device" \
-	| xargs -r -n 1 batctl interface add
-
+	for device in $(ubus call network.interface dump | jsonfilter -e "@.interface[@.proto='gluon_mesh' && @.up=true].device"); do
+		batctl interface add "$device"
+		local device_sanitized="$(echo "$device" | sed 's/[^a-zA-Z0-9]/_/g')"
+		local hop_penalty="$(ubus call network.interface dump | jsonfilter -e "@.interface[@.device='$device'].data.hop_penalty")"
+		if [ -n "$hop_penalty" ] && [ "$hop_penalty" -ne 0 ]; then
+			batctl hardif "$device" hop_penalty "$hop_penalty"
```

Comment:
This should set the hop penalty unconditionally, so changing it back from non-zero to zero will work (not really relevant to Gluon, as we will never change the state of a configured interface, but simpler and more correct)